sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CryptographicKeyShouldNotBeTooShort.cs

```diff
@@ -24,11 +32,468 @@ public void Method()
 
 
             const int localInvalidSize = 1024;
-            new RSACryptoServiceProvider(1024); // Noncompliant {{Use a key length of at least '2048' bits.}}
+            new RSACryptoServiceProvider(1024); // Noncompliant {{Use a key length of at least 2048 bits for RSA cipher algorithm.}}
 //          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
             new RSACryptoServiceProvider(1024, new CspParameters()); // Noncompliant
             new RSACryptoServiceProvider(invalidKeySizeConst); // Noncompliant
             new RSACryptoServiceProvider(localInvalidSize); // Noncompliant
         }
     }
+
+    public class SystemSecurityCryptographyDSA
+    {
+        public void DefaultConstructors()
+        {
+            var dsa1 = new DSACng(); // Compliant - default is 2048
+            var dsa2 = new DSACryptoServiceProvider(); // Noncompliant - default is 1024
+            var dsa3 = new DSAOpenSsl(); // Compliant - default is 2048
+        }
+
+        public void CompliantKeySizeConstructors(CspParameters parameters)
+        {
+            var dsa1 = new DSACng(3072);
+            var dsa2 = new DSAOpenSsl(2048);
+        }
+
+        public void NonCompliantKeySizeConstructors(CspParameters parameters)
+        {
+            var dsa1 = new DSACng(512); // Noncompliant {{Use a key length of at least 2048 bits for DSA cipher algorithm.}}
+            var dsa2 = new DSACryptoServiceProvider(512); // Noncompliant {{Use a key length of at least 2048 bits for DSA cipher algorithm.}}
+            var dsa3 = new DSACryptoServiceProvider(1024, parameters); // Noncompliant
+            var dsa4 = new DSACryptoServiceProvider(3072); // Noncompliant - this is not valid
+            var dsa5 = new DSACryptoServiceProvider(2048, parameters); // Noncompliant - this is not valid
```

Comment:
I agree. But I think it is overkill to add an addtional message for a case that we should actually never encounter (since this is not a valid case). So, for simplicity purpose, I'd suggest to keep it as is

Reply:
Ok, considering that the message asks to replace the entire class.